horace_core/sqw/@Experiment/set_instrument.m

```diff
@@ -35,7 +35,7 @@
     obj.instruments = instr_or_fun;
 
 elseif isa(instr_or_fun,"function_handle")
-    if numel(varargin)==1 && iscell(varargin{1})
+    if isscalar(instr_or_fun) && iscell(varargin{1})
```

Comment:
```suggestion
    if isscalar(varargin) && iscell(varargin{1})
```
checking

Reply:
No, this would not work. 
The function signatrue is:
`obj = set_instrument(obj,instr_or_fun,varargin)` so the suggestion would break the code